### PR TITLE
Update docs to reflect support for form onReset

### PR DESCRIPTION
**Pull request by aesopwolf**
This is a small change to the docs to reflect the fact that `onReset` can be handled by [Form Events](https://facebook.github.io/react/docs/events.html#form-events)

### CODE REVIEW

File: docs/_js/examples/todo.js [hunk 1]
```diff
@@ -3,18 +3,21 @@ class TodoApp extends React.Component {
   constructor(props) {
     super(props);
     this.handleChange = this.handleChange.bind(this);
+    this.handleReset = this.handleReset.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
-    this.state = {items: [], text: ''};
+    this.originalValues = {items: [], text: ''};
+    this.state = this.originalValues;
```
Comment (Kritika2808) [inline docs/_js/examples/todo.js]: Just curious, why {items: [], text: ''} is first assigned to this.originalValues and then this.state
Why this change was made ?

Comment (aesopwolf) [inline docs/_js/examples/todo.js]: `this.originaValues` is used in the `handleReset` method. I can use a different approach, or add an comment describing the behavior.

**Comment by bvaughn** [comment]
Thank you for filing this PR!

I'm sorry to be the bearer of bad news, but the documentation and source code for [reactjs.org](https://reactjs.org/) now lives in a different repository: [reactjs/reactjs.org](github.com/reactjs/reactjs.org). (For more info on why we made this move, see issue #11075.)

If you'd like to re-open this PR on the new repo we'll be happy to review and continue discussion there!
